Creation Of Management Controller And View

GuenoleCHERUBIN opened the pull request:
Creation of the management controller
Creation of the management folder in view
Creation of the accounts view in management folder

GuenoleCHERUBIN committed:
Creation Of Management Controller And View

Creation of the manaagement controller
Creation of the management folder in view
Creation of the accounts view in management folder

GuenoleCHERUBIN committed:
Update On Accounts And Controllers

Changed modify-account button to a link
Added management to the roles in MainController
Creation of actionModifyAccount
Added the link url to the rules

GuenoleCHERUBIN committed:
Creation Of Modify Account Form Model And View

Creation of the model for the modify account form
Update of the management controller to have the modify action
Creation of thee view where the form is displayed

GuenoleCHERUBIN committed:
Creation Of The Update Account Function

Creation of the private function updateAccount
This function can't change role and password
And shouldn't be able to do it
Update of the modifyAccount view

alexisljn commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+
+class ManagementController extends MainController
+{
+    /**
```
PHPDOC Missing

alexisljn commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+    /**
+     * Display a page where the accounts are shown
+     */
+    public function actionAccounts($id = null)
```
The method name should be more descriptive

GuenoleCHERUBIN replied:
But it's more logical to call for "accounts" in the URL (example-site.com/management/accounts), than calling for "showAccounts" (example-site.com/management/show-accounts).

alexisljn replied:
According to Yii2 routes pattern, it seems that you're right

alexisljn commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+        $user =  User::findOne(['id' => $id]);
+        //dd($user);
+        $form = new ModifyAccountForm();
+        if($form->load($_POST) && $form->validate()) {
```
Why don't you use the ```Yii::$app->request->post()```

GuenoleCHERUBIN replied:
Why though ? The two are equivalent.

alexisljn replied:
Yes, it is just to keep the framework logical. But it's not indispensable.

alexisljn commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+
+    private function actionModifiyAccount($id)
+    {
+        $user =  User::findOne(['id' => $id]);
```
You have to check if the User exists in DDB, otherwise you have to throw a 'UserNotFoundException'

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+
+use app\controllers\mainController\MainController;
+use app\models\databaseModels\User;
+use Yii;
```
Unused import

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+
+
+
+
```
Remove this spaces

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,78 @@
+
+        } else {
+            echo "save failed.";
+        }
```
Unnecessary if/else statement: use
`if(!$user->save()) {
    // Do something if save failed
}` instead

EmileCalixte commented on models/ModifyAccountForm.php:
```diff
@@ -0,0 +1,34 @@
+
+namespace app\models;
+
+use Yii;
```
Unused import

EmileCalixte commented on models/ModifyAccountForm.php:
```diff
@@ -0,0 +1,34 @@
+/**
+ * ModifyAccountForm is the model behind the modify account form.
+ */
+
```
Remove blank line between PHPDoc and class declaration

EmileCalixte commented on models/ModifyAccountForm.php:
```diff
@@ -0,0 +1,34 @@
+    /**
+     * @return array the validation rules.
+     */
+
```
Remove blank line between PHPDoc and fnuction declaration

GuenoleCHERUBIN committed:
Had The PHP Doc In Management Controller

Haded the PHP doc in the management controller

GuenoleCHERUBIN committed:
Change PHP Doc For Management Controller

Changed the PHP Doc of the management controller
so it's more accruate

GuenoleCHERUBIN committed:
Hade A Check For User Existence

Hadded a private function to check if the user
exist

GuenoleCHERUBIN committed:
Remove Unnecessary Spaces

Removed unnecessary spaces in modify account form
and management controller

GuenoleCHERUBIN committed:
Changed A Missundertanding In Management Controller

Changed a missunderstanding i had on the variable
from the functions

GuenoleCHERUBIN committed:
Hade The Creation Of A New Account

Hadded the form to creat a new account
Hadded the roles on main controller for the action
create account
Hadded the two functions to create the acount
in the management controller
Hadded the view to create account

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,140 @@
+     * @param int $id
+     * @return string
+     */
+    private function actionModifiyAccount($id)
```
Please add role edition in this action

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,140 @@
+        $user->is_active    = $form->is_active;
+
+        if(!$user->save()){
+            echo "save failed.";
```
Don't write "save failed", throw a CannotSaveException instead:
```throw new CannotSaveException($user);```

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,140 @@
+        $user->lastname     = $form->lastname;
+        $user->email        = $form->email;
+        $user->is_validated = "0";
+        $user->is_active = "0";
```
You can simply write `$user->is_validated = false;`, same for is_active

EmileCalixte commented on controllers/ManagementController.php:
```diff
@@ -0,0 +1,140 @@
+        $user->role         = $form->role;
+
+        if(!$user->save()){
+            echo "save failed.";
```
Please throw a CannotSaveException

EmileCalixte commented on models/ModifyAccountForm.php:
```diff
@@ -0,0 +1,31 @@
+    public $firstname;
+    public $lastname;
+    public $email;
+    public $is_validated;
```
It's not logical to have is_validated here: it suggests that the admin can edit this "is_validated" value, whereas it's set as true when the user has set his password

EmileCalixte commented on models/ModifyAccountForm.php:
```diff
@@ -0,0 +1,31 @@
+            [['firstname','lastname', 'email', 'is_validated', 'is_active'], 'required'],
+            // email has to be a valid email address
+            ['email', 'email'],
+        ];
```
String max lengths are missing

EmileCalixte commented on views/management/createAccount.php:
```diff
@@ -0,0 +1,41 @@
+use yii\bootstrap\ActiveForm;
+
+$this->title = 'Create Account';
+$this->params['breadcrumbs'][] = $this->title;
```
You can remove this line, the breadcrums widget will be disabled.

GuenoleCHERUBIN committed:
Add Role To Modify Account Form

Added the role to the modify account form
the modify account action and to the view

Changed values in createAccount in the management controller
to be more accurate
$user->is_active was meant to be set as true

GuenoleCHERUBIN committed:
Add Validation Of The Roles

Added the validation of the roles in ModifyAccountForm
deleted the is_validated in the modifyAccount action

alexisljn commented on views/management/accounts.php:
```diff
@@ -0,0 +1,67 @@
+/** @var \yii\data\ActiveDataProvider $usersDataProvider */
+
+echo $usersDataProvider->count;
+//dd($usersDataProvider);
```
useless dd

GuenoleCHERUBIN committed:
Merge remote-tracking branch 'origin' into management

Resolve Conflict

Resolved conflicts in MainController

GuenoleCHERUBIN committed:
Added Throw Exception When Failed Save

Added throw exceptions when the save to database fail

alexisljn commented on views/management/modifyAccount.php:
```diff
@@ -0,0 +1,50 @@
+use yii\bootstrap\ActiveForm;
+
+$this->title = 'Modify Account';
+$this->params['breadcrumbs'][] = $this->title;
```
Remove this line, breadcrumbs widget will be disabled

GuenoleCHERUBIN committed:
Finish PHP Doc And Suppressed Useless dd()

Added php doc for the throw exception
Removed commented dd()
